nats-server/configuration/securing_nats/tls.md

```diff
@@ -59,3 +59,90 @@ tls: {
+Think carefully about revocation and cycling times, as well as automation, when picking a CA. 
+If arbitrary applications inside your organization need to connect, use a cert from your in-house CA. 
+If only resources inside a specific environment need to connect, that environment might have its own dedicated automatic CA, eg in Kubernetes clusters, so use that.
+
+**Only** for **testing** purposes does it makes sense to generate self signed certificates, even your own CA. 
+This is a **short** guide on how to do just that and what to watch out for.
+
+> **DO NOT USE these certificates in production!!!**
+
+### Problems With Self Signed Certificates
+
+#### Missing in Relevant Trust Stores 
+
+As they should, these are **not trusted** by the system your server or clients are running on.
+
+One option is to specify the CA in every client you are using.
+In case you make use of `verify` and `verify_and_map` you need to specify `ca_file` in the server.
+If you are having a more complex setup involving cluster, gateways or leaf nodes, `ca_file` needs to be present in `tls` maps used to connect to the server with self signed certificates. 
+While this works for server and libraries from the NATS eco system, you will experience issues when connecting with other tools such as your Browser.
+
+Another option is to configure your system's trust store to include self signed certificate(s).
+Which trust store needs to be configured depends on what you are testing.
+* This may be your OS for server and certain clients.
+* The runtime environment for other clients like Java or Python.
+* Your browser for monitoring endpoints and websockets.
+
+Please check your system's documentation on how to trust a particular self signed certificate.
+
+#### Missing Subject Alternative Name 
+
+Another common problem is failed [identity validation](https://tools.ietf.org/html/rfc6125).
+The IP or DNS name to connect to needs to match a [Subject Alternative Name (SAN)](https://tools.ietf.org/html/rfc4985) inside the certificate.
+Meaning, if a client/browser/server connect via tls to `127.0.0.1`, the server needs to present a certificate with a SAN containing the IP `127.0.0.1` or the connection will be closed with a handshake error.
+
+### Creating Self Signed Certificates for Testing
+
+The simplest way to generate a CA as well as client and server certificates is [mkcert](https://github.com/FiloSottile/mkcert).
+This zero config tool generates and installs the CA into your **local** system trust store(s) and makes providing SAN straight forward. 
+Here is an example:
+
+Generate a CA and output the location of the root CA cert file `rootCA.pem`. 
+Next generate a certificate, valid for use by `localhost` and the IP `::1`(`-cert-file` and `-key-file` overwrite default file names).
+Then start a nats server using the generated certificate.
+
+```bash
+mkcert -install
+mkcert -CAROOT
```

Comment:
If I am not mistaken, this simply displays the location of where the CA root cert will be stored. Unless you have the example where you pass `--tls_cacert` to the `nats-server` exe, then I am not sure you need to have that.

Reply:
Verify does just that... I also mentioned that ca_file defaults to the system store.  
I will move mkcert-CAROOT further down and turn that into a separate step.

Reply:
Moved it into a separate section saying:
Examples in this document make use of the certificates generated so far.
To simplify examples using the CA certificate, copy `rootCA.pem` into the same folder where the certificates were generated. To obtain the CA certificate's location use this command: